src/core/src/op/scatter_elements_update.cpp

```diff
@@ -16,6 +16,31 @@
 using namespace ngraph;
 using namespace std;
 
+namespace {
+void validate_scatter_element_types(const ov::Node* op) {
+    const element::Type& data_et = op->get_input_element_type(0);
+    const element::Type& indices_et = op->get_input_element_type(1);
+    const element::Type& updates_et = op->get_input_element_type(2);
+    const element::Type& axis_et = op->get_input_element_type(3);
+
+    NODE_VALIDATION_CHECK(op,
+                          indices_et.is_integral(),
+                          "Indices element type must be integral_number, but is: ",
+                          indices_et);
+
+    NODE_VALIDATION_CHECK(op, axis_et.is_integral(), "Axis element type must be integral_number, but is: ", axis_et);
+
```

Comment:
One thing is that not all data types are supported in case of reduction mode. For example, what does it mean `sum` reduction for boolean type

Reply:
In my opinion, boolean type shouldn't be a problem for ``sum``, ``prod``, ``min``, ``max`` and it is supported case by frameworks, additional info can be added to the spec:
* For ``boolean`` type of ``data`` input, *reduction* ``sum``, ``prod`` behaves like logical ``OR``, ``AND`` accordingly, but there is no implementation for ``mean``.

Reply:
`mean` mode is not clear for all integer types.:)
@tomdol, @mitruska, can you please investigate behaviors of different frameworks for these conner cases?

Reply:
The `mean` reduction type is only present in pytorch and it always rounds down the calculated value to the nearest integer. We've added this reduction type to OV to cover this case in PT FE and I think it should be implemented accordingly.